src/Text.php

```diff
@@ -34,7 +56,9 @@ public function getHtml(): string
      */
     public function addParagraph($text)
     {
-        $this->content .= $this->getApp()->getTag('p', [], $text);
+        $this->content[] = (new HtmlValue())->dangerouslySetHtml('<p>');
+        $this->content[] = (new HtmlValue())->set($text);
+        $this->content[] = (new HtmlValue())->dangerouslySetHtml('</p>');
```

Comment:
here you could simply use `$this->dangerouslyAddHtml('</p>');`, but that's additional function call, so probably better not.